### ISSUE TITLE
VFPT should provide a quickfix action which removes trailing whitespace

### ISSUE THREAD

**Issue by forki** [fsprojects/VisualFSharpPowerTools#1130]
FsLint squigglies should squiggle over full error

![image](https://cloud.githubusercontent.com/assets/57396/10215396/73befeb8-6821-11e5-89c3-a71acedb6a33.png)

**Comment by vasily-kirichenko** [comment]
That's what it did initially and the squiggles spread over many lines in most lint cases so the majority of lines in file were touched. So I artificially cut the span to one line, two chars at most. Maybe we should add a setting for this?

**Comment by forki** [comment]
nope. If this intentionally then it's OK. I change my request to: VFPT should provide a quickfix action which removes trailing whitespace.
Note: this is different from normal "Format code" since I don't always want to touch non-whitespace code.

**Comment by vasily-kirichenko** [comment]
Yes, I think quick fixes should be provided for most Linter hints.

**Comment by dungpa** [comment]
It would be great to have relevant info (input ranges, suggestions, etc.) from FSharpLint to implement quick fixes /cc @duckmatt. I bet it's really tricky for certain rules; but who knows.

**Comment by forki** [comment]
Let's start with this "simple case" as a test.

**Comment by duckmatt** [comment]
A simple non-formatted suggestion could probably be generated for lots of the lint hint warnings as an extension of the lint api, this could then be run through fantomas to give a nice fix

**Comment by dungpa** [comment]
We may be able to start with simple quick fixes once FSharpLint integration completes.

**Comment by cloudRoutine** [comment]
ideally the lint fixes would similar to C# refactorings/fixes

![](https://i1.visualstudiogallery.msdn.s-msft.com/2ddb7240-5249-4c8c-969e-5d05823bcb89/image/file/169875/1/fig10.png)

**Comment by vasily-kirichenko** [comment]
I think we should implement simple actions first, without preview, then, if anyone still has energy and time, we could add the preview.

**Comment by dsyme** [comment]
Slightly off topic: I'm spending some time working with a finance industry customer. I just upgraded VFPT in that context and the "end of line whitespace warnings" came on by default.  I found this felt wrong: I've personally never cared about end-of-line whitespace in 30 years of programming, so for me these warnings are spurious, and that's likely to be the case for other people except where people are using specific editors that (mistakenly IMHO) remove whitespace by default.

So I recommend this warning be off by default (or FSharpLint should be off by default, which I've suggested for possible performance reasons).  It seems wrong for beginners to be asked to have an opinion about this setting and to then have to dig into options and disable something - the warning squiggles are intrusive enough that the >50% who don't like the setting on by default will have to go turn it off. And it seems wrong that I'm turning off FSharpLint just to get rid of these warnings (I know I don't have to turn off FSharpLint, but it's the quickest way to make the warnings go away). 

thx
